kv/dist_sender.go

```diff
@@ -429,6 +431,13 @@ func (ds *DistSender) Send(call *client.Call) {
 	for {
 		reply := call.Reply
 		err := util.RetryWithBackoff(retryOpts, func() (util.RetryStatus, error) {
+			select {
+			case <-ds.closer:
+				call.Reply.Header().SetGoError(util.Errorf("sender is closed"))
```

Comment:
I'm not convinced we want to return an error to the client like this. Better to just close off the connection to the client when this happens so client retries. This sort of an error would propagate back through the kv gateway, through http and up to the application. We need a specific retry error of some sort.

Add a TODO here to return an error indicating retry.